app/Http/Controllers/GitHubSocialiteController.php

```diff
@@ -63,27 +63,26 @@ public function callback()
 
     protected function retrieveUser($gitHubUser): User
     {
-        if (session('auth-user-email')) {
+        if (session('auth-user-id')) {
             /*
              * If there already was a local user created for the email used
              * on GitHub, then let's use that local user
              */
-            return User::where('email', session('auth-user-email'))->first();
+            return User::find(session('auth-user-id'))->first();
         }
 
-        if ($user = User::where('github_id', $gitHubUser->id)->orWhere('email', $gitHubUser->email)->first()) {
```

Comment:
Nu verliezen we wel de find op `github_id`, Github email is niet altijd gelijk aan de Spatie account email

Reply:
Die if-condition is enkel voor users die wél dezelfde email gebruiken voor Spatie & GitHub. De find op GitHub id gebeurt hieronder via `firstOrCreate`